src/DynamoCoreWpf/ViewModels/Preview/WatchViewModel.cs

```diff
@@ -274,5 +275,41 @@ public void CountLevels()
         {
             Levels = maxListLevel > 0 ? Enumerable.Range(1, maxListLevel).Reverse().Select(x => x).ToList() : Enumerable.Empty<int>();
         }
+
+        /// <summary>
+        /// Get the NodeLabel for this WatchViewModel and any children.
+        /// Each level of children will be indented by 2 spaces.
+        /// </summary>
+        /// <param name="model">The WatchViewModel to process.</param>
+        /// <param name="depth">The number of levels of indentation.</param>
+        /// <param name="includeKey">If true the list or dictionary key will be included in the string.</param>
+        /// <returns></returns>
+        public static string GetNodeLabelTree(WatchViewModel model, int depth = 0, bool includeKey = false)
```

Comment:
Does this need to be static?

Wondering if we on the `PreviewControl` could do this
``` csharp
private void OnCopyToClipboardClick(object sender, RoutedEventArgs e)
{
    string content = cachedLargeContent.GetNodeLabelTree();
    if (!string.IsNullOrEmpty(content)) Clipboard.SetText(content);
}
```
and on the Watch node do this
```csharp
private void OnCopyToClipboardClick(object sender, RoutedEventArgs e)
{
    string content = rootWatchViewModel.GetNodeLabelTree();
    if (!string.IsNullOrEmpty(content)) Clipboard.SetText(content);
}

Reply:
I did wonder about this but it's a recursive function so it needs to be able to call itself with a child WatchViewModel as the argument:
https://github.com/StudioLE/Dynamo/blob/ac4c8ffaea0eb0be5020746eadc2cadec9380862/src/DynamoCoreWpf/ViewModels/Preview/WatchViewModel.cs#L300

Reply:
We could possibly do something along the lines an extension method. That might allow both use cases?

```c#
public static string GetNodeLabelTree(this WatchViewModel model, int depth = 0, bool includeKey = false)
```

Reply:
Would this not work, or am i missing something?
```csharp
IEnumerable<string> labels = model.Children.Select(x => x.GetNodeLabelTree(depth + 1, true));
```

Reply:
You're right, I've got no idea what I was thinking

Reply:
Fixed in d11ed33